libs/qrcode/index.php

```diff
@@ -1,12 +1,43 @@
 <?php 
+include '../php-commons/XSSHelpers.php';
 include 'qrlib.php';
 $cacheDir = 'cache/';
 
 $encode = 'Invalid Input.';
 if(isset($_GET['encode'])) {
+    $input = XSSHelpers::xssafe($_GET['encode']); // first pass scrub and character encoding enforcement
+
     $len = strlen($_GET['encode']);
-    if($len > 0 && $len < 4000) {
-        $encode = $_GET['encode'];
+    if($len > 0 && $len < 4000) { // check QR container limits
+        $encode = $input;
+    }
+
+    // TODO: Replace this with server config variable
+    $HTTP_HOST = '';
+    $matches = array();
+    switch($_SERVER['HTTP_HOST']) {
+        case 'leaf.va.gov':
+            $HTTP_HOST = 'leaf.va.gov';
+            break;
+        case 'leaf-preprod.va.gov':
+            $HTTP_HOST = 'leaf-preprod.va.gov';
+            break;
+        case 'leaf-dev.va.gov':
+            $HTTP_HOST = 'leaf-dev.va.gov';
```

Comment:
We can't have these domain names in a public repo. We can pull in globals.php and compare HTTP_HOST instead

Reply:
Updated. Note that the update will potentially break QR-related functionality for some sites, but it's more important to get this fix merged. The breaking functionality is accepted.